### ISSUE TITLE
Tooltip disappearing on hover difficult for screen mag users

### ISSUE THREAD

**Issue by anevins12** [ZoeBijl/apg-tooltip#2]
While meeting WCAG, this pattern is still difficult to use for screen magnify users. The issues:

- The tooltip is unintentionally revealed when panning around the page
- The tooltip is unintentionally dismissed when panning outside of the scripted corridors

Would you consider implementing the tooltip on click? So that the tooltip is triggered on click/ SPACE keys and dismissed using the ESC key (or clicked outside). That won't, however, resolve the issue of unintentionally triggering tooltips.

**Comment by Malvoz** [comment]
Are these problems also present on pages with HTML `title`?

I think the show on hover VS show on click is essentially describing the distinctions between tooltips (i.e. HTML `title` ) VS popups (e.g. Open UI [`<popup>`](https://open-ui.org/components/popup)).

> - The tooltip is unintentionally revealed when panning around the page

This could perhaps be mitigated by introducing a small delay, as suggested in https://www.w3.org/TR/wai-aria-1.1/#tooltip:

> authors **_SHOULD_** display the tooltip after a short delay

> Typical tooltip delays last from one to five seconds.

**Comment by anevins12** [comment]
> Are these problems also present on pages with HTML title?

No, the behavior is much worse. HTML title attributes are not available for keyboard-only users and the tooltip content is not persistent on hover. 

> I think the show on hover VS show on click is essentially describing the distinctions between tooltips (i.e. HTML title ) VS popups (e.g. Open UI [<popup>](https://open-ui.org/components/popup)).

I'm not familiar with `<popup>` elements, but perhaps the tooltip pattern should be rethought if its target audience excludes low vision users.

Edit: I shouldn't say that it excludes low vision users. I understand the persistent on hover intends to make sure people with enlarged pointers can still read the content. Still, meeting WCAG doesn't mean the tooltip is useable.

**Comment by Malvoz** [comment]
> > Are these problems also present on pages with HTML title?
>
> No, the behavior is much worse. HTML title attributes are not available for keyboard-only users and the tooltip content is not persistent on hover.

There are browser bugs for keyboard accessible tooltips (HTML `title`, SVG `<title>`), for example: https://bugs.chromium.org/p/chromium/issues/detail?id=829352.

> perhaps the tooltip pattern should be rethought

This tooltip pattern should hopefully closely match the behavior of the aforementioned tooltips (if they were fully accessible), per the original issue: https://github.com/w3c/aria-practices/issues/127. Therefore I don't think there's too much room in changing the conventional behavior.
